src/logsumexp.jl

```diff
@@ -24,9 +26,32 @@ The result is computed using a single pass over the data.
 # References
 
 [Sebastian Nowozin: Streaming Log-sum-exp Computation](http://www.nowozin.net/sebastian/blog/streaming-log-sum-exp-computation.html)
+
+See also [`logsumexp!`](@ref)
 """
 logsumexp(X::AbstractArray{<:Number}; dims=:) = _logsumexp(X, dims)
 
+"""
+$(SIGNATURES)
+
+Compute `out .= log.(sum!(out, exp.(X)))` in a numerically stable way that avoids
+intermediate over- and underflow.
+
+The result is computed using a single pass over the data.
```

Comment:
```suggestion
Compute `log.(sum!(r, exp.(X); dims=dims))`, summing over the singleton dimensions of `r`,
and overwriting `r` with the results.

The result is computed using a single pass over the data, in a numerically stable way that avoids intermediate
over- and underflow.
```

I think the `out .= ...` part is confusing. It's more clear to say explicitly that `r` is overwritten.
Also the docstring should mention that the sum is carried out over singleton dimensions of `r`.
Compare to the docstring of `sum!`.

Reply:
> Also the docstring should mention that the sum is carried out over singleton dimensions of out.

That's already implied by `sum!` (which BTW does not support a `dims` keyword argument).

Reply:
I updated the docstring.